mne/preprocessing/tests/test_ica.py

```diff
@@ -608,4 +610,58 @@ def test_fit_params():
     ICA(fit_params=fit_params)  # test no side effects
     assert_equal(fit_params, {})
 
+
+@requires_sklearn
+def test_bad_channels():
+    """Test if exception is raised when unsupported channels are used.
+    Test for raw and evoked data"""
+    chs = [i for i in _kind_dict]
+    data_chs = _DATA_CH_TYPES_SPLIT + ['eog']
+    chs_bad = list(set(chs) - set(data_chs))
+    info = create_info(len(chs), 500, chs)
+    data = np.random.rand(len(chs), 50)
+    raw = RawArray(data, info)
+    data = np.random.rand(100, len(chs), 50)
+    epochs = EpochsArray(data, info)
+
+    n_components = 0.9
+    ica = ICA(n_components=n_components)
+
+    for inst in [raw, epochs]:
+        for ch in chs_bad:
+            # Test case for only bad channels
+            picks_bad1 = pick_types(inst.info, exclude='bads',
+                                    **{str(ch): True})
+            # Test case for good and bad channels
+            picks_bad2 = pick_types(inst.info, exclude='bads',
+                                    **{str(ch): True})
+            assert_raises(ValueError, ica.fit, inst, picks=picks_bad1)
+            assert_raises(ValueError, ica.fit, inst, picks=picks_bad2)
+        assert_raises(ValueError, ica.fit, inst, picks=[])
+
+
+@requires_sklearn
+def test_eog_channel():
+    """Test that EOG channel is included when performing ICA"""
+    raw = Raw(raw_fname, preload=True)
+    events = read_events(event_name)
+    picks = pick_types(raw.info, meg=True, stim=True, ecg=False,
+                       eog=True, exclude='bads')
+    epochs = Epochs(raw, events, event_id, tmin, tmax, picks=picks,
+                    baseline=(None, 0), preload=True)
+    n_components = 0.9
+    ica = ICA(n_components=n_components)
+    # Test case for MEG and EOG data. Should have EOG channel
+    for inst in [raw, epochs]:
+        picks1 = pick_types(inst.info, meg=True, stim=False, ecg=False,
+                            eog=True, exclude='bads')
+        ica.fit(inst, picks=picks1)
+        assert_true(any('EOG' in ch for ch in ica.ch_names))
+    # Test case for MEG data. Should have no EOG channel
+    for inst in [raw, epochs]:
+        picks1 = pick_types(inst.info, meg=True, stim=False, ecg=False,
+                            eog=False, exclude='bads')
+        ica.fit(inst, picks=picks1)
+        assert_false(any('EOG' in ch for ch in ica.ch_names))
```

Comment:
how long does it take to run the 2 added tests? Can you make it even faster? cropping the raw, using fastica?